pkg/agent/client_test.go

```diff
@@ -355,7 +354,15 @@ func newClosePacket(connID int64) *client.Packet {
 
 func goleakVerifyNone(t *testing.T, options ...goleak.Option) {
 	t.Helper()
-	if err := goleak.Find(options...); err != nil {
+	var err error
+	waitErr := wait.PollImmediate(100*time.Millisecond, 3*time.Second, func() (bool, error) {
```

Comment:
LGTM, I was going to make a similar change. We should really make a testing library, and move this there. I'm happy to do that in a follow-up PR.

Reply:
Actually, the only problem with this approach is `t.Helper()` is no longer effective, since `wait.Poll` adds a few more functions on the stack. I think it might be worth using the manual for-sleep approach instead.

Reply:
Nevermind! It matters from the context where `t.Error` is called, which is in the same function as `t.Helper`, so you can disregard this.